### DISCUSSION TITLE
Export Error

### DISCUSSION THREAD

**Discussion by mingyuecall** [marp-team/marp#343] [Q&A] [answered]
Failure to export: [CLIError] Failed converting Markdown. (Unexpected '!'. Escaping special characters with \ may help.)

Hello folks! I can be sure that there are no extra "!" in the slides.

Thank you!

**Comment by yhatt** [comment, marked as answer]
You can find [this error is caused by `postcss-selector-parser`](https://stackoverflow.com/questions/68050244/unexpected-escaping-special-characters-with-may-help-angular-12-build-d) [by googling the error message](https://www.google.com/search?q=Unexpected+%27!%27.+Escaping+special+characters+with+%5C+may+help). Are not an invalid character `!` is containing in CSS selectors?

Check whether inline styles and theme CSS using in your Markdown slide are valid markup. W3C CSS validator is useful for debugging your stylesheets: https://jigsaw.w3.org/css-validator/

#### Minimum reproducible example

```markdown
---
marp: true
---

👇 This CSS is invalid so the slide background would not become red.

<style>
section! {
  background: red;
}
</style>
```

**Reply by mingyuecall** [reply]
Thank you so much!
